### PR TITLE
Feature/rp 273 improve placement of score view buttons

### PR DESCRIPTION
**Pull request by esotrope**
show more

### CODE REVIEW

File: webapp/src/main/webapp/src/app/assessments/results/assessment-results.component.html [hunk 1]
```diff
@@ -1,41 +1,24 @@
...
+    <sb-radio-group
+      *ngIf="scoreTypeOptions != null && scoreTypeOptions.length > 1"
+      [options]="scoreTypeOptions"
+      [(ngModel)]="scoreType"
+      id="score-type"
+      buttonGroupStyles="btn-group-sm"
+      buttonStyles="btn-default"
+      analyticsCategory="AssessmentResults"
+      analyticsEvent="ChangeScoreView"
+    ></sb-radio-group>
```
Comment (esotrope) [inline webapp/src/main/webapp/src/app/assessments/results/assessment-results.component.html]: reduced markup noise by using css grid

File: webapp/src/main/webapp/src/app/assessments/results/assessment-results.component.ts [hunk 2]
```diff
@@ -542,7 +537,11 @@ export class AssessmentResultsComponent implements OnInit {
       this.subjectDefinition != null ? this.calculateStats() : null;
 
     // compute table when ready
-    if (this.subjectDefinition != null && this.exams != null) {
+    if (
+      this.exams != null &&
+      this.subjectDefinition != null &&
+      this.subjectDefinition.alternateScore != null
```
Comment (esotrope) [inline webapp/src/main/webapp/src/app/assessments/results/assessment-results.component.ts]: NPE bugfix oops

File: webapp/src/main/webapp/src/app/assessments/results/view/results-by-student/results-by-student.component.ts [hunk 3]
```diff
@@ -153,14 +153,18 @@ export class ResultsByStudentComponent implements OnInit {
...
+      ...(this.subjectDefinition.alternateScore != null
+        ? createAlternateScoreColumns(
+            this.assessment.alternateScoreCodes,
+            this.subjectDefinition.alternateScore.codes
+          )
+        : []),
+      ...(this.subjectDefinition.claimScore != null
+        ? createClaimColumns(
+            this.assessment.claimCodes,
+            this.subjectDefinition.claimScore.codes
```
Comment (esotrope) [inline webapp/src/main/webapp/src/app/assessments/results/view/results-by-student/results-by-student.component.ts]: more NPE bugfixes x l

File: webapp/src/main/webapp/src/app/assessments/results/assessment-results.component.ts (hunk 2 above)
Comment (malaffoon) [reply webapp/src/main/webapp/src/app/assessments/results/assessment-results.component.ts]: Seems odd to check that value here: the call only requires `this.exams` and `this.subjectDefinition`. Shouldn't the `toScoreTable` method be responsible for dealing with a null alternateScore?

Comment (esotrope) [reply webapp/src/main/webapp/src/app/assessments/results/assessment-results.component.ts]: hm. probably a philosophical battle but i always feel like the method caller should be responsibly for knowing whether it needs to call the method or not. This is trying to say "if there are alt scores make a table for them"

**Pull request merged**